src/event-stream/event-server.ts

```diff
@@ -997,20 +997,20 @@ export function parseNewBlockMessage(
 ) {
   const counts = newCoreNoreBlockEventCounts();
 
+  // Nakamoto blocks now include their own `block_time`, but this will be empty for pre-Nakamoto
+  // blocks. We'll use the parent burn block timestamp as the receipt date for those. If both are
+  // blank, there's something wrong with Stacks core.
+  const block_time = msg.block_time ?? msg.burn_block_time;
+  if (!block_time) {
```

Comment:
if `block_time == 0` the error will be thrown. Isn't more appropriate to explicitly checks for both null and undefined, but not falsy values like 0? Like `if (blockTime == null) {...}`

Reply:
Hah you read my mind! Just added this